src/cbang/openssl/SSLContext.cpp

```diff
@@ -291,8 +323,120 @@ void SSLContext::loadSystemRootCerts() {
+    bool valid;
+
+    cert = (SecCertificateRef) CFArrayGetValueAtIndex(anchors, i);
+
+    summary0 = SecCertificateCopySubjectSummary(cert);
+    if (summary0) {
+      summary = getStringFromCFString(summary0);
+      CFRelease(summary0);
+    }
+
+    LOG_DEBUG(5, "Cert " << i << ": " << summary);
+
+    // validate root cert
+    // anchor certs are already trusted, but can be expired, revoked, or
+    // explicitly distrusted by a user
+    // on macOS 10.11 and earlier, all the root certs will have expired
+    valid = false;
+
+    SecTrustResultType result;
+    SecTrustRef trust = NULL;
+    SecPolicyRef x509Policy = SecPolicyCreateBasicX509();
+    if (x509Policy) {
+      err = SecTrustCreateWithCertificates(cert, x509Policy, &trust);
+      if (err == errSecSuccess)
+        err = SecTrustSetPolicies(trust, x509Policy);
+
+      // OPTIONAL, apparently, for our use
+      // set params for validating a root cert
+      // see ValidateSystemTrust in
+      // https://github.com/macports/macports-ports/blob/master/security/certsync/files/certsync.m
+
+      if (err == errSecSuccess) {
+        if (__builtin_available(macOS 10.14, *)) { // macOS 10.14+
+          valid = SecTrustEvaluateWithError(trust, NULL);
+          LOG_DEBUG(5, "Called SecTrustEvaluateWithError()");
+        } else {
+          err = SecTrustEvaluate(trust, &result); // macOS 10.3–10.15
+          LOG_DEBUG(5, "Called SecTrustEvaluate()");
+          if (err == errSecSuccess) {
+            switch (result) {
+              case kSecTrustResultUnspecified:
+              case kSecTrustResultProceed:
+                valid = true;
+                break;
+              default:
+                valid = false;
+                break;
+            }
+          }
+        }
+      }
+
+      if (x509Policy) CFRelease(x509Policy);
+      if (trust) CFRelease(trust);
+    }
+
+    if (valid) {
+      LOG_DEBUG(5, "Trusting certificate: " << summary);
+      // convert and add cert to trusted store
+      data = SecCertificateCopyData(cert);
+      if (data) {
+        const uint8_t *bytes = CFDataGetBytePtr(data);
+        long len = CFDataGetLength(data);
+        X509 *xcert = d2i_X509(0, &bytes, len);
+
+        if (!xcert) {
+          LOG_WARNING("Error parsing root cert: " << SSL::getErrorStr());
+          CFRelease(data);
+          continue;
+        }
+
+        if (!X509_STORE_add_cert(verifyStore, xcert)) {
+          auto error = ERR_get_error();
+
+          if (ERR_GET_REASON(error) != X509_R_CERT_ALREADY_IN_HASH_TABLE) {
+            LOG_ERROR("Error adding root cert: " << SSL::getErrorStr(error));
+            X509_free(xcert);
+            CFRelease(data);
+            break;
+          }
+        }
+        trustedCount++;
+        X509_free(xcert);
+        CFRelease(data);
+      } else {
+        LOG_ERROR("Failed to get data for certificate: " << summary);
+      }
+    } else {
+      LOG_DEBUG(3, "Untrusted certificate: " << summary);
+    }
```

Comment:
This is a nitpick but, these can be flattened to one line.  I.e.

```
      } else LOG_ERROR("Failed to get data for certificate: " << summary);
    } else LOG_DEBUG(3, "Untrusted certificate: " << summary);
```

Reply:
Done.